src/ctype/CType.chain.ts

```diff
@@ -26,7 +26,10 @@ export async function store(
   > = await blockchain.api.tx.ctype.add(ctype.hash)
   const txStatus: TxStatus = await blockchain.submitTx(identity, tx)
   if (txStatus.type === 'Finalised') {
-    ctype.owner = identity.address
+    txStatus.payload = {
+      ...ctype,
+      owner: identity.address,
+    }
   }
   return txStatus
```

Comment:
If the transaction isn't finalized, then the `txSatus` doesn't have a `payload` property.    
Are we not accessing `payload` and using it somewhere else in the codebase, which could potentially crash if `payload` is `undefined` ?  

(this is just a thought exercise, since I'm guessing that case was covered already in the previous implementation)

Reply:
yeah, this code is weird. if it is not finalized, it will actually throw an error. We have to refactor it here.

Reply:
https://github.com/KILTprotocol/sdk-js/issues/149